src/components/Products/ProductListingView/ProductListing.jsx

```diff
@@ -0,0 +1,38 @@
+import './productListing.css';
+import {useFilter} from '../../../contexts/filterContext';
+import {SidebarFilter} from '../SidebarFilterView/SidebarFilter';
+import { ProductCard } from '../ProductCardView/ProductCard';
+import {
+    getSortedData,
+    getProductsInPriceRange,
+    getCateogrisedProducts,
+    getDatainStarRatingRange
+  } from '../../../utils/sortAndFilter/index';
+
+
+const ProductListing = () => {
+
+const {state : {sortBy, categories, priceRange, products, starRating}} = useFilter();
+const sortedProducts = getSortedData(products, sortBy);
+const productsInPriceRange = getProductsInPriceRange(sortedProducts, priceRange);
+const productsInStarRating = getDatainStarRatingRange(productsInPriceRange, starRating);
+const filteredProducts = getCateogrisedProducts(productsInStarRating, categories);
```

Comment:
can do all the four filter in a custom hook and then put it in the filterReducer file and then import it, make your code cleaner (maybe).
This looks good to go according to your preference